### PR TITLE
feat(user): generate and delete sample data

### PR DESCRIPTION
**Pull request by apoorvdixit88**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Added support to generate and delete sample data for user

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
Sample data is dummy data which user can use to generate dummy payments/refunds and test behaviour of analytics, etc.

## How did you test it?
Use the following curls to generate and delete sample data:
Generate:
```
curl --location 'http://localhost:8080/user/sample_data' \
--header 'Content-Type: application/json' \
--header 'Cookie: token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoiMWZjNzgyOTUtM2JiMy00OGJjLThlZDgtNzFjMzVjMzMxYWU2IiwibWVyY2hhbnRfaWQiOiJtZXJjaGFudF8xNjk3NzE2ODcwIiwicm9sZV9pZCI6Im1lcmNoYW50X2FkbWluIiwiZXhwIjoxNjk3ODg5NjcxLCJvcmdfaWQiOiJvcmdfejQ5ZExMeTdmbllUODN5TDY3clEifQ.CJzEQ2qbhn-qUiiVBSdvCJiLvWp-5wCF9R54gth6QbQ' \
--header 'Authorization: Bearer JWT' \
--data '{
    "record": 20
}'

```
DB entries can be checked for payment intents, attempts and refunds.

Delete:
```
curl --location --request DELETE 'http://localhost:8080/user/sample_data' \
--header 'Content-Type: application/json' \
--header 'Cookie: token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoiMWZjNzgyOTUtM2JiMy00OGJjLThlZDgtNzFjMzVjMzMxYWU2IiwibWVyY2hhbnRfaWQiOiJtZXJjaGFudF8xNjk3NzE2ODcwIiwicm9sZV9pZCI6Im1lcmNoYW50X2FkbWluIiwiZXhwIjoxNjk3ODg5NjcxLCJvcmdfaWQiOiJvcmdfejQ5ZExMeTdmbllUODN5TDY3clEifQ.CJzEQ2qbhn-qUiiVBSdvCJiLvWp-5wCF9R54gth6QbQ' \
--header 'Authorization: Bearer JWTI' \
--data '{    
}'
```

Data from tables will be deleted successfully.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/routes/app.rs [hunk 1]
```diff
@@ -771,13 +771,31 @@ pub struct User;
...
-        web::scope("/user")
-            .app_data(web::Data::new(state))
-            .service(web::resource("/signin").route(web::post().to(user_connect_account)))
-            .service(web::resource("/signup").route(web::post().to(user_connect_account)))
-            .service(web::resource("/v2/signin").route(web::post().to(user_connect_account)))
-            .service(web::resource("/v2/signup").route(web::post().to(user_connect_account)))
-            .service(web::resource("/change_password").route(web::post().to(change_password)))
+        let mut route = web::scope("/user").app_data(web::Data::new(state));
+
+        #[cfg(feature = "olap")]
```
Comment (racnan) [inline crates/router/src/routes/app.rs:776]: is this required ?

File: crates/router/src/routes/app.rs [hunk 2]
```diff
@@ -771,13 +771,31 @@ pub struct User;
...
+        #[cfg(feature = "dummy_connector")]
+        {
+            route = route
+                .service(
+                    web::resource("/sample_data/generate")
+                        .route(web::post().to(generate_sample_data)),
+                )
+                .service(
+                    web::resource("/sample_data/delete")
+                        .route(web::delete().to(delete_sample_data)),
```
Comment (racnan) [inline crates/router/src/routes/app.rs:795]: rename this, no need to add `generate` and `delete` in route

**Review by ThisIsMani — DISMISSED** [review]

File: crates/router/src/routes/app.rs [hunk 3]
```diff
@@ -842,7 +843,15 @@ impl User {
...
-            .service(web::resource("/role/{role_id}").route(web::get().to(get_role)))
+            .service(web::resource("/role/{role_id}").route(web::get().to(get_role)));
+
+        #[cfg(feature = "dummy_connector")]
+        {
+            route = route
+                .service(web::resource("/sample_data").route(web::post().to(generate_sample_data)))
+                .service(web::resource("/sample_data").route(web::delete().to(delete_sample_data)))
+        }
+        route
```
Comment (racnan) [inline crates/router/src/routes/app.rs:854]: good to have: these can be written without repeating the route name.

**Review by racnan — APPROVED** [review]
minor thing.

**Review by ThisIsMani — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**